src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -314,16 +315,29 @@ impl Service<Vec<Event>> for CloudwatchLogsSvc {
 
     fn call(&mut self, req: Vec<Event>) -> Self::Future {
         if self.token_rx.is_none() {
-            let events = req
+            // We need to take into account the flight time from here to aws servers,
+            // so we are reducing timestamp acceptable range from both sides
+            // for this amount.
+            let buffer_time = Duration::minutes(1);
+            let now = Utc::now();
+            // Acceptable range of Event timestamps.
+            let age_range = (now - Duration::days(14) + buffer_time).timestamp_millis()
+                ..(now + Duration::hours(2) - buffer_time).timestamp_millis();
+
+            let mut events = req
                 .into_iter()
                 .map(|mut e| {
                     self.encoding.apply_rules(&mut e);
                     e
                 })
                 .map(|e| e.into_log())
                 .map(|e| self.encode_log(e))
+                .filter(|e| age_range.contains(&e.timestamp))
                 .collect::<Vec<_>>();
 
+            // Sort by timestamp
+            events.sort_by_key(|e| e.timestamp);
```

Comment:
True. I missed this even though the same optimization was applied to the 24/h filter.

Reply:
After working on this for a bit, I don't think it's worth it. We can't avoid iterating since we need to transform `Event` to `InputLogEvent`. While yes we could almost completely avoid cost of the filter, it's just 2 integer comparisons and `&&` for each event, we would need to add additional complexity which would have a negative performance for smaller event counts, while when we do have something to filter out from the start of the `Vec` we would need to shift everything else. While the current solution, when we do have something to filter, has benefit of reducing number of events to be sorted. 

All in all, I would say it's a blurry tie on performance side, but simplicity is on the side of the current solution.